deployment.go

```diff
@@ -72,7 +74,38 @@ func (r *marathonClient) Deployments() ([]*Deployment, error) {
 	if err != nil {
 		return nil, err
 	}
-
+	// Allows loading of deployment steps from the Marathon v1.X API
+	// Implemented as a Quick fix for issue https://github.com/gambol99/go-marathon/issues/153
+	if r.config.V1X {
```

Comment:
I'd rather not have this be a switch you need to enable first. Maybe it's possible to unmarshal both pre-1.1 and 1.1 versions and pick the one we find populated? This would also allow for a smoother transition on the consumer end if upstream decides to go back to the pre-1.1 format.